app/models/service_template.rb

```diff
@@ -388,10 +388,8 @@ def create_resource_actions(ae_endpoints)
   end
 
   def self.create_from_options(options)
-    create(options.except(:config_info, :picture).merge(:options => { :config_info => options[:config_info] })).tap do |tmp|
-      if options[:picture]
-        tmp.update_attributes!(:picture => Picture.create(:content => options[:picture][:content], :extension => options[:picture][:extension]))
-      end
+    create!(options.except(:config_info, :picture).merge(:options => { :config_info => options[:config_info] })).tap do |tmp|
+      tmp.update_attributes!(:picture => create_picture(options))
```

Comment:
maybe this extraction of the create_picture method should pass `options[:picture]` since that's all you need in it.  It should simplify the above ^